common/middleware/set-services.js

```diff
@@ -0,0 +1,13 @@
+const services = require('../services')
+
+const setServices = (req, res, next) => {
```

Comment:
The original suggestion for this had a `servicesCache` in here. I left it out only because I was not 100% sure it would be safe when we do end up passing the req through. The [suggested approach](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/783cf9797e93d64757f32a1789c570ff4eff566e/common/middleware/set-services.js) looked like it would potentially cache a service instantiated with some req and then wouldn't update it when a different req came along, which would cause all sorts of data problems.

Very happy to for someone to explain why I'm wrong, it's easily added back in.

Reply:
The services cache was within the middleware so only lived for the life cycle of the request so what you feared might happen, wouldn't ;)

Reply:
The previous code allowed for the instantiation of the service if and only when it was actually needed, using the `req` object in the closure within `setServices`. NB. you'll have to address how to ensure that services called by the serviceified services also get the `req` context passed on to them.

Reply:
Yeah, once all services have been refactored in some fashion to take on the req, it will be possible to pass it on from anywhere that calls the service. Maybe. Looking back, that seems like a bold statement.